appveyor.yml

```diff
@@ -1,22 +1,58 @@
 version: 1.0.{build}
+os:
+- Visual Studio 2013
+# only run PR-CA on the VS 2015 OS - please change the scripts if this string changes
+- Visual Studio 2015
+
+install:
+  - git submodule update --init --recursive
+  - ps: |
+      Add-Type -AssemblyName System.IO.Compression.FileSystem
+      if (!(Test-Path -Path "C:\maven" )) {
+        (new-object System.Net.WebClient).DownloadFile(
+          'http://www.us.apache.org/dist/maven/maven-3/3.2.5/binaries/apache-maven-3.2.5-bin.zip',
+          'C:\maven-bin.zip'
+        )
+        [System.IO.Compression.ZipFile]::ExtractToDirectory("C:\maven-bin.zip", "C:\maven")
+      }
+  - cmd: SET PATH=C:\maven\apache-maven-3.2.5\bin;%JAVA_HOME%\bin;%PATH%
+
+cache:
+  - C:\sonarqube\downloads -> appveyor.yml
+  - C:\users\appveyor\.m2 -> appveyor.yml
+
```

Comment:
Caching the Maven cache and the downloads directory where the SonarQube server zip files are downloaded. Caching takes ~30s, while downloading varies quite a lot - ultimately there is a small gain with caching. 

Note that the zip files are versioned, e.g. SonarQube_server_5.2.zip so there is no change of conflicts.